hdinsight-with-edge-node/azuredeploy.json

```diff
@@ -0,0 +1,402 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "location": {
+            "type": "string",
+            "allowedValues": [ 
+                "Central US",
+                "East Asia",
+                "East US",
+                "Japan East",
+                "Japan West",
+                "North Europe",
+                "South Central US",
+                "Southeast Asia",
+                "West Europe",
+                "West US"
+                ],
+            "metadata": { 
+                "description": "The location where all azure resources will be deployed."
+            }
+        },
+        "vnetName": {
+            "type": "string",
+            "metadata": { 
+                "description": "The name of the virtual network that will be created and have the cluster and edge node deployed into."
+            }
+        },
+        "vnetAddressSpace": {
+            "type": "string",
+            "defaultValue": "10.0.0.0/16",
+            "metadata": { 
+                "description": "The virtual network's address range in CIDR notion. It must be contained in one of the standard private address spaces: 10.0.0.0/8, 172.160.0.0/12, or 192.168.0.0/16."
+            }
+        },
+        "vnetSubnetName": {
+            "type": "string",
+            "defaultValue": "default",
+            "metadata": { 
+                "description": "The name of the subnet to create in the virtual network."
+            }
+        },
+        "vnetSubnetAddressRange": {
+            "type": "string",
+            "defaultValue": "10.0.0.0/24",
+            "metadata": { 
+                "description": "The subnet's address range in CIDR notation. It must be contained by the address space of the virtual network."
+            }
+        },
+        "clusterName": {
+            "type": "string",
+            "metadata": { 
+                "description": "The name of the HDInsight cluster to create."
+            }
+        },
+        "clusterLoginUserName": {
+            "type": "string",
+            "defaultValue": "admin",
+            "metadata": { 
+                "description": "These credentials can be used to submit jobs to the cluster and to log into cluster dashboards."
+            }
+        },
+        "clusterLoginPassword": {
+            "type": "securestring",
+            "metadata": { 
+                "description": "The password for the cluster login."
+            }
+        },
+        "sshUserName": {
+            "type": "string",
+            "defaultValue": "hdiuser",
```

Comment:
Please let the user pick the username rather than have a publicly exposed default value for this